core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -590,12 +593,59 @@ private void checkMerkleRoot() throws VerificationException {
         }
     }
 
+    private void checkSegwitCommit() throws VerificationException {
+        if (transactions == null)
+            throw new VerificationException("No transactions in block");
+        final Transaction coinBase = transactions.get(0);
+        byte[] rootHash = null;
+        for (TransactionOutput out : coinBase.getOutputs()) {
+            final byte[] pkScript = out.getScriptBytes();
+            if (pkScript.length < 38) continue;
+            final byte[] expected = HEX.decode("6a24aa21a9ed");
```

Comment:
It is less magic than it seems:

```
6a - OP_RETURN
24 - push 36 bytes to stack
aa21a9ed - commitment header
```

In the pull request for Bitcoin Core these constants are set in code as opposed to symbols.

https://github.com/bitcoin/bitcoin/pull/8149/files#diff-7ec3c68a81efff79b6ca22ac1f1eabbaR3511

It's always the same no matter which network.